phpBB/includes/functions_admin.php

```diff
@@ -1872,7 +1887,6 @@ function sync($mode, $where_type = '', $where_ids = '', $resync_parents = false,
 						}
 					}
 				}
-				unset($post_info);
```

Comment:
This shouldn't be removed. The current placement ensures that the post info array is both created and deleted inside step 5 and therefore stays encapsulated. If `$post_ids` is empty the compact in your event will throw a PHP notice with `Notice: Undefined variable: post_info`

Reply:
@marc1706 I'm unsure how to resolve this one.  [In the event within my extension here,](https://github.com/toxyy/anonymousposts/blob/master/event/listener.php#L547) I'm appending data to` forum_data` from `post_info` which requires a new fieldname as well.  I could separate this into two events, one to append to `forum_data` and one to append to the fieldnames, but since they are both related, it seemed better to do it in one event as it's meaningless to add a key onto fieldnames without corresponding data from `forum_data` or vice versa.  Would it be better to make a separate variable for this, or add a line
```php
if( !isset($post_info)
{
    $post_info = array()
}
```
or `$post_info = isset($post_info) ?: array();`

before the `sync_modify_forum_data` event.

Which would be the better solution?